### ISSUE TITLE
mergetool support in server extension

### ISSUE THREAD

**Issue by wotanii** [jupyter/nbdime#323]
I love the new diff-tool support in the jupyter frontend (#297)

Is it planned to add merge-tool support to the jupyter frontend as well?

**Comment by vidartf** [comment]
There is no current plans, but it should be possible. What kind of usage pattern to you envision for this? That is, what would be the steps for using this? How was the merge triggered? etc.

**Comment by wotanii** [comment]
I want my users to do all basic git-operations in the notebook. I'm working on our own plugin, that adds buttons for push/commit/add/pull/etc. NbDime is (currently) used to show the diff to head. The only thing missing is the possibility to resolve a merge conflict (which might happen after a pull). 

So in my scenario, the conflict is triggered by my own extension via git operations. Then I'd call nbdime's merge-handler with pointers to local, base and remote. NbDime should write the merge-result to the local file (or maybe to a new file?), and then my plugin creates a merge-commit with the new file. 

So basically I want to be able to call `nbmerge-web` the same way I can call `nbdiff-web` from the notebook.

**Comment by vidartf** [comment]
As long as you have the nbdime merge driver installed, any merge conflicts should result in a valid notebook with conflict markers (which can be opened like a normal notebook). So that can be used as a workaround for now (although it might be hard to determine when it should be marked as resolved).

If I understand your use case correctly then, you only need to serve up the merge page, but do not need any buttons directly in the UI (at least, not supplied by nbdime). That should be feasible.

PS: You might be interested in looking at the https://github.com/jupyterlab/jupyterlab-git project that is making a similar git extension for Jupyterlab (still unreleased).

**Comment by wotanii** [comment]
thanks. This solves my issue. In fact the merge-driver is even better than what I was thinking about
